addons/test_mail/tests/test_mail_composer.py

```diff
@@ -994,32 +994,80 @@ def test_mail_composer_delete(self):
 
     @users('employee')
     @mute_logger('odoo.tests', 'odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
-    def test_mail_composer_document_based(self):
-        """ Tests a document-based mass mailing with the same address mails
-        This should be allowed and not considered as duplicate in this context
+    def test_mail_composer_duplicates(self):
+        """ Ensures emails sent to the same recipient multiple times
+            are only sent when they are not duplicates
         """
-        self.test_records.write({
-            'customer_id': False,
-            'email_from': 'duplicate.email@test.example.com',
-        })
         self.template.write({
-            'auto_delete': False,  # keep sent emails to check content
-            'email_to': '{{ object.email_from }}',
+            'subject': 'Common Subject',
+            'body_html': '<p>Common Body</p>',
+            'email_to': 'test@test.lan',
             'partner_to': '',
+            'auto_delete': False,
         })
+        # Guarantee points of variation
+        self.test_records[0].write({'name': 'A'})
+        self.test_records[1].write({'name': 'B'})
         # launch composer in mass mode
         composer_form = Form(self.env['mail.compose.message'].with_context(
             self._get_web_context(self.test_records, add_web=True,
                                   default_template_id=self.template.id)
         ))
         composer = composer_form.save()
+        # no difference
         with self.mock_mail_gateway(mail_unlink_sent=False), self.mock_mail_app():
-            composer.with_context(mailing_document_based=False)._action_send_mail()
+            composer._action_send_mail()
         self.assertEqual(len(self._mails), 1, 'Should have sent 1 email, and skipped a duplicate.')
+        self._mails = []
+        composer._onchange_template_id_wrapper()
```

Comment:
Seems this could be removed (perform the onchange after updating the template) ?